File: ABC/274/C++/D.cpp
#include<iostream>
#include<vector>
#include<numeric>
#define rep(i,n) for(int i=0; (i)<(n); ++i)
using namespace std;

bool isAchievable(vector<int> v, int t, bool isY);

int main(){
    // int N = 3; int x = -1; int y = 1;
    // vector<int> A_e = {1}; vector<int> A_o{2,3};
    int N, x, y; cin >> N >> x >> y;
    vector<int> A_e(N/2); vector<int> A_o((N+1)/2);
    rep(i,N) (i)%2==0 ? cin>>A_o[i/2] : cin>>A_e[i/2];
    (isAchievable(A_o, x, false) && isAchievable(A_e, y, true)) ? cout<<"Yes" : cout<<"No";
    cout << endl;
    return 0;
}

bool isAchievable(vector<int> v, int t, bool isY){
    int L = accumulate(v.begin(), v.end(), 0);
    int N = v.size();
    vector<vector<bool>> DP(N, vector<bool>(2*L+1, false));
    DP[0][ v[0]+L] = true;
    DP[0][-v[0]+L] = isY;
    for(int r=1; r<N; ++r){
        for(int c=0; c<=2*L; ++c){
            if(0<=c-v[r]   && DP[r-1][c-v[r]]) DP[r][c] = true; 
            if(c+v[r]<=2*L && DP[r-1][c+v[r]]) DP[r][c] = true;
        }
    }
    return -L<=t && t<=L && DP[N-1][t+L];
}

File: ABC/274/Julia/D.jl
function isAchievable(Arr::Array{Int}, t::Int, isY::Bool)
    L = sum(Arr)
    N = length(Arr)
    #Dynamic Programming matirx.
    DP = zeros(Int, 2L+1, N)
    DP[Arr[1]+L+1, 1] = 1
    if isY # p1, X direction, must be (A1,0), cannot be (-A1,0)
        DP[-Arr[1]+L+1, 1] = 1
    end
    # DP
    for col = 2:N
        a = Arr[col]
        for row = 1:2L+1
            if get(DP, (row-a, col-1), 0) == 1
                DP[row, col] = 1
            end
            if get(DP, (row+a, col-1), 0) == 1
                DP[row, col] = 1
            end
        end
    end
    
    return (get(DP,(t+L+1, N), 0) == 1)
end
function main()
    N, x, y = parse.(Int, readline()|>split)
    A = parse.(Int, readline()|>split)
    # A_e = {Ai | Ai ∈ A and i ∈ even}
    A_e = Array{Int}([])
    # A_e = {Ai | Ai ∈ A and i ∈ odd}
    A_o = Array{Int}([])
    for (i, a) ∈ enumerate(A)
        i%2==0 ? push!(A_e, a) : push!(A_o, a)
    end
    ( isAchievable(A_o, x, false) && isAchievable(A_e, y, true)) ? "Yes"|>print : "No"|>print
end
main()
